python/lsst/ts/scheduler/driver/feature_scheduler.py

```diff
@@ -656,3 +737,88 @@ def _get_survey_name_from_observation(self, observation):
         """
         # For now simply return the "notes" field.
         return observation["note"][0].split(":")[0]
+
+    def _get_driver_target_from_observation_data_frame(
+        self, observation_data_frame: typing.Tuple[pandas.Timestamp, pandas.Series]
+    ) -> FeatureSchedulerTarget:
+        """Convert an observation data frame into a FeatureSchedulerTarget.
+
+        Override super class method to deal with Feature Scheduler Target
+        custom requirements.
+
+        Parameters
+        ----------
+        observation_data_frame :  tuple [pandas.Timestamp, pandas.Series]
+            An observation data frame.
+
+        Returns
+        -------
+        target : FeatureSchedulerTarget
+            Feature scheduler target.
+        """
+
+        fbs_observation = self._get_fbs_observation_from_observation_data_frame(
+            observation_data_frame
+        )
+
+        (
+            observing_script_name,
+            observing_script_is_standard,
+        ) = self.get_survey_observing_script(
+            self._get_survey_name_from_observation(fbs_observation)
+        )
+
+        target = FeatureSchedulerTarget(
+            observing_script_name=observing_script_name,
+            observing_script_is_standard=observing_script_is_standard,
+            observation=np.array(fbs_observation, ndmin=1),
+            log=self.log,
+            **self.script_configuration,
+        )
+
+        return target
+
+    def _get_fbs_observation_from_observation_data_frame(
+        self, observation_data_frame: typing.Tuple[pandas.Timestamp, pandas.Series]
+    ) -> np.ndarray:
+        """Convert observation pandas data fram to an fbs observation.
+
+        Parameters
+        ----------
+        observation_data_frame :  tuple [pandas.Timestamp, pandas.Series]
+            An observation data frame.
+
+        Returns
+        -------
+        fbs_observation : np.ndarray
+            Feature scheduler observation.
+        """
+        fbs_observation = empty_observation()
+
+        for key in self.fbs_observation_named_parameter_map():
```

Comment:
for-loops are perfectly OK of course but this could be replaced with a list comprehension. Lines 807 to 809 as well. Up to you if you want to change that.